packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -620,13 +620,25 @@ abstract class ExternalApplicationListener extends Resource implements IApplicat
    *
    * It's possible to add conditions to the TargetGroups added in this way.
    * At least one TargetGroup must be added without conditions.
+   *
+   * NOTE: if you are deploying new infrastructure using this method,
+   * it is recommended to set the `removeRuleSuffixFromLogicalId: false`
+   * property so that the logicalId for the generated `ListenerRule`
+   * is consistent with other methods for managing `ListenerRule`s.
+   *
+   * If you have already deployed a `ListenerRule` using this method
+   * and need to migrate to using the more feature rich `addAction()`
+   * method, then you will need to set the `removeRuleSuffixFromLogicalId: true`
+   * property there to avoid having CloudFormation attempt to replace your resource.
    */
   public addTargetGroups(id: string, props: AddApplicationTargetGroupsProps): void {
     checkAddRuleProps(props);
 
     if (props.priority !== undefined) {
+      const removeRuleSuffixFromLogicalId = props.removeRuleSuffixFromLogicalId ?? true;
```

Comment:
I'm not sure I see a reason to have the option to add `Rule` at all to this function, unless you think that users might want to switch back from `addAction()` to `addTargetGroups()`, but from your descriptions it sounds like `addAction()` is just better. 

I realize that making them all consistent was totally my suggestion, but I now that I'm seeing it implemented, I think I was wrong. If a user wants to migrate back this way, they can just add `Rule` to the end of the `id`.

Reply:
This addresses the problem of a user wanting to use `addTargetGroup()` while having a consistent logicalId. I... don't think most people would ever discover this. The interesting question is how to move forward with consistent logiacalIds. I'm pretty sure that the Right Way to do this is to change the default and provide a FeatureFlag to revert to legacy behaviour.

TBH, I'm not particular so long as we unblock the ability for `addAction()` to deal with the inconsistency.